contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -110,11 +102,14 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
    *  - adminFee: fee that will be paid to the Router owner for operating the network
    */
   function _updateConfig(bytes memory config) internal override {
-    (uint96 adminFee, bytes4 handleOracleFulfillmentSelector, uint32[] memory maxCallbackGasLimits) = abi.decode(
-      config,
-      (uint96, bytes4, uint32[])
-    );
-    s_config = Config({
+    (
+      uint16 maxConsumers,
+      uint96 adminFee,
+      bytes4 handleOracleFulfillmentSelector,
+      uint32[] memory maxCallbackGasLimits
+    ) = abi.decode(config, (uint16, uint96, bytes4, uint32[]));
+    s_config = SubscriptionConfig({
```

Comment:
Maybe keep the config in the Router, like it was before, and only call setMaxConsumers() on the subscription contract in every _updateConfig()?

Reply:
Or have an internal virtual getter (my preference)